components/VueLayerMarker.vue

```diff
@@ -24,7 +37,54 @@ export default defineComponent({
         src: {
             type: String,
             default: "/marker.png"
+        },
+        selectSrc: {
+            type: String,
+            default: "/marker.png"
+        },
+        details: { // ToDo - define positionData object like index.vue (or re-use that one by referencing it, somehow)
+          type: Object,
+          default: () => {}
         }
+    },
+    emits: ["popup-toggled"],
+    data () {
+      return {
+        iconSrc: this.src,
+        selectediconSrc: this.selectSrc
+      };
+    },
+    methods: {
+      onSelect (e : any) {
```

Comment:
```suggestion
      onSelect (e : SelectEvent) {
```

Good practice: avoid use of `any` as much as possible, that's the point of using TypeScript.
Change above is not enough, you will also need to import it I think (not tested).

Reply:
Resolved by adding a comment about the weird nature of the VueLayers event